### DISCUSSION TITLE
Keda - Scaler - Service Bus

### DISCUSSION THREAD

**Discussion by mineshp100** [kedacore/keda#2195] [Q&A / Need Help]
I have a question in regards to the service bus scaler for Keda.

https://keda.sh/docs/1.4/scalers/azure-service-bus/

In the link above, it says the queuelength (messagecount) is "Amount of active messages in your Azure Service Bus queue or topic to scale on."

We tested this and from what i can see or observe, message count is how many messages can each pod can handle.

So for example, if we have 4 pods in a deployment and in our keda config we set the messagecount to be 10, Keda will only autoscale if the messages on the servicebus queue is greater than 40 messages. (10 * 4). 

But the description says otherwise.

Could someone please confirm what the message count means?

**Comment by tomkerkhove** [comment]
We no longer maintain v1.4 and recommend upgrading to 2.4

**Reply by mineshp100** [reply]
Hi Tom,

Apologies - I pasted the wrong version in the link above.

I changed the version and the description is still the same. 

We are currently using 2.3.2.

**Comment by tomkerkhove** [comment]
Is the above correct @ahmelsayed? As far as I see, it's just scaling out to be able to coop with that. If I have 1 message, it sometimes even goes to 4 instances.

**Comment by amirschw** [comment]
> We tested this and from what i can see or observe, message count is how many messages can each pod can handle.
So for example, if we have 4 pods in a deployment and in our keda config we set the messagecount to be 10, Keda will only autoscale if the messages on the servicebus queue is greater than 40 messages. (10 * 4).

This is indeed the current behavior. It is because the Service Bus scaler, like all other scalers (except for the cpu/memory ones), uses the `AverageValue` metric type. For this reason, the currentMetricValue is computed by taking the average of the given metric across all Pods in the HorizontalPodAutoscaler's scale target (taken from the [HPA docs](https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/#algorithm-details)).

I just sent a [PR](https://github.com/kedacore/keda/pull/2309) to add support for the `Value` metric type in addition to `AverageValue`. In your example, when `Value` will be used, it will autoscale when the message count in the queue/topic will be greater than 10.
